# Joint state controller
joint_state_controller:
  publish_rate: 500
  type: joint_state_controller/JointStateController

# Effort based controller
arm_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
     - shoulder_pan_joint
     - shoulder_lift_joint
     - elbow_joint
     - wrist_1_joint
     - wrist_2_joint
     - wrist_3_joint
  constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      shoulder_pan_joint: {trajectory: 0.1, goal: 0.1}
      shoulder_lift_joint: {trajectory: 0.1, goal: 0.1}
      elbow_joint: {trajectory: 0.1, goal: 0.1}
      wrist_1_joint: {trajectory: 0.1, goal: 0.1}
      wrist_2_joint: {trajectory: 0.1, goal: 0.1}
      wrist_3_joint: {trajectory: 0.1, goal: 0.1}
  gains:
    shoulder_pan_joint:  {p: 20000, d: 10, i: 10, i_clamp: 1}
    shoulder_lift_joint: {p: 20000, d: 10, i: 10, i_clamp: 1}
    elbow_joint:   {p: 10000, d: 10, i: 5, i_clamp: 1}
    wrist_1_joint: {p: 5000,  d: 1,  i: 0, i_clamp: 1}
    wrist_2_joint: {p: 5000,  d: 1,  i: 0, i_clamp: 1}
    wrist_3_joint: {p: 1000,  d: 1,  i: 0, i_clamp: 1}
  stop_trajectory_duration: 0.5
  state_publish_rate:  25
  action_monitor_rate: 10

  velocity_ff:
      shoulder_pan_joint: 1.0
      shoulder_lift_joint: 1.0
      elbow_joint: 1.0
      wrist_1_joint: 1.0
      wrist_2_joint: 1.0
      wrist_3_joint: 1.0

#Individual controllers
shoulder_pan_joint:
  type: effort_controllers/JointPositionController
  joint: shoulder_pan_joint
  pid: {p: 20000, d: 10, i: 10}

shoulder_lift_joint:
  type: effort_controllers/JointPositionController
  joint: shoulder_lift_joint
  pid: {p: 20000, d: 10, i: 10}

elbow_joint:
  type: effort_controllers/JointPositionController
  joint: elbow_joint
  pid: {p: 10000, d: 10, i: 5}

wrist_1_joint:
  type: effort_controllers/JointPositionController
  joint: wrist_1_joint
  pid: {p: 5000, d: 1, i: 0.0}

wrist_2_joint:
  type: effort_controllers/JointPositionController
  joint: wrist_2_joint
  pid: {p: 5000, d: 1, i: 0.0}

wrist_3_joint:
  type: effort_controllers/JointPositionController
  joint: wrist_3_joint
  pid: {p: 1000, d: 1, i: 0.0}
